### PR TITLE
Xdm/products

### PR DESCRIPTION
**Pull request by dancingcactus**
## Additional Details
Provides and example of how and implementation instructions for an XDM schema.

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: what-to-implement/commerce.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+* [cartAbandons](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmcartabandons) (_Optional_) -- A cart is no longer accessible or purchasable by the user.
+* [checkouts](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmcheckouts) (_Highly Recommended_) -- Set when a user is no longer browsing for products but is in the process of purchasing a product
+* [productListAdds](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmproductlistadds) (_Highly Recommended_) -- When a products is added to the list. Be sure to set the product in the `productListItems` at the same time.
+* [productListOpens](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmproductlistopens) (_Optional_) -- When a new product list is created. E.g. a new shopping cart is created.
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: * "E.g." can't be at the start of a sentence. I'd suggest: "When a new product list is created (e.g., a new shopping cart)."

File: what-to-implement/commerce.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+{% hint %}
+
+* [productViews](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmproductviews) (_Highly Recommended_) -- a view of the product occured. Be sure to set the product viewed in the `productListItems`
+* [cartAbandons](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmcartabandons) (_Optional_) -- A cart is no longer accessible or purchasable by the user.
+* [checkouts](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmcheckouts) (_Highly Recommended_) -- Set when a user is no longer browsing for products but is in the process of purchasing a product
+* [productListAdds](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmproductlistadds) (_Highly Recommended_) -- When a products is added to the list. Be sure to set the product in the `productListItems` at the same time.
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: * products -> product

File: what-to-implement/commerce.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+* [productListOpens](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmproductlistopens) (_Optional_) -- When a new product list is created. E.g. a new shopping cart is created.
+* [productListRemovals](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmproductlistremovals) (_Highly Recommended_)
+* [productListReopens](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmproductlistreopens) (_Optional_)-- When a custoemr is re-activated by the user. This often happens via re-marketing campaigns
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: * Typo: customer

File: what-to-implement/commerce.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+* [purchases](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmpurchases) (_Highly Recommended_)-- An order is accepted. Must have a product list
+* [saveForLaters](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmsaveforlaters) (_Optional_) -- When a product is saved for future use.
+
+Here is an example of how you would set these `Measures` in the SDK.
+
+```javascript
+alloy("event", {
+  "xdm": {
+    "commerce":{
+      "poductViews":{
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: * Typo: `productViews`

File: what-to-implement/commerce.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+* [productViews](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmproductviews) (_Highly Recommended_)-- A single product is viewed
+* [purchases](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmpurchases) (_Highly Recommended_)-- An order is accepted. Must have a product list
+* [saveForLaters](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/context/commerce.schema.md#xdmsaveforlaters) (_Optional_) -- When a product is saved for future use.
+
+Here is an example of how you would set these `Measures` in the SDK.
+
+```javascript
+alloy("event", {
+  "xdm": {
+    "commerce":{
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: Space after colons for consistency (this applies to the other code samples as well).

File: what-to-implement/commerce.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+      "poductViews":{
+        "value":1
+      }
+    }
+  }
+});
+```
+
+The commerce object also has a special field for collecting order details. This is called `order` and it is used to collect information about an order.
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: The two sentences seem redundant. Maybe "The order field is an object used to collect information about an order." or something like that.

File: what-to-implement/commerce.md [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+  * [currencyCode](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/data/order.schema.md#xdmcurrencycode) (_optional_)-- the [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217) currency for this payment method.
+  * [paymentAmount](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/data/paymentitem.schema.md#xdmpaymentamount) (_Highly Recommended_) -- The value of the payment in the currency code specified.
+  * [paymentType](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/data/paymentitem.schema.md#xdmpaymenttype) (_Highly Recommended_) -- The type of payment (e.g. `credit_card`, `gift_card`, `paypal`). See the list of [known values](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/data/paymentitem.schema.md#xdmpaymenttype-known-values) for details
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: * Comma after "e.g.".

File: what-to-implement/commerce.md [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+  * [transactionID](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/data/paymentitem.schema.md#xdmtransactionid) (_optional_) -- A unique ID for this payment transaction.
+* [priceTotal](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/data/order.schema.md#xdmpricetotal) (_Highly Recommended_) -- The total for this order after all discounts and taxes have been applied.
+* [pruchaseID](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/data/order.schema.md#xdmpurchaseid) (_Highly Recomended_) -- The unique identifier assigned by the seller for this purchase.
+* [purcahseOrderNumer](_optional_) -- A unique identifier assigned by the purchaser for this purchase.
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: * Typo: `purchaseOrderNumber`

File: what-to-implement/commerce.md [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+
+Here is an example of a typical purchase in Alloy.
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: We've tried to avoid the term "Alloy" in the docs (except in code examples), since it's more of a code name. We usually say "Here is an example of a typical purchase using the SDK.".

File: what-to-implement/commerce.md [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+        "qauntity":1
+      }
+    ]
+  }
+});
+```
+
+## Lists of Products
+
+The product list intended to be sent whenever the above items are set to indicate what products the actions are happening to. It is a list of [productListItems](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/content/productlistitem.schema.md). Each product has a number of optional fields.
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: * The product list intended -> The product list is intended
* Would it make sense to say "indicate which products correspond to the actions" or something like that? If not, that's fine. It's just a little awkward to end with "what products the actions are happening to".

File: what-to-implement/commerce.md [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+
+* [SKU](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/content/productlistitem.schema.md) (_Highly Recommended_) -- Store Keeping Unit. It is the unique identifier for the product.
+* [currencyCode](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/content/productlistitem.schema.md#xdmcurrencycode) (_optional_) -- the [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217) currency for for the product. This is only useful when you can have products with different currency codes and when it applies like when there is a purcahse or add to cart.
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: * applies like when -> applies, for example, when
* typo "purchase"

File: what-to-implement/commerce.md [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+* [productName](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/content/productlistitem.schema.md#xdmname) (_Highly Recommended_) -- This is set to the display name or human readable name of the product.
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: * human readable -> human-readable

File: what-to-implement/commerce.md [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+* [priceTotal](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/content/productlistitem.schema.md#xdmpricetotal) (_Highly Recommended_) -- Should only be set when applicable. For example, It may not be possible to set on `productView` since different variations of the product can have different prices but on an `productListAdds`.
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: * It -> it
* on an `productListAdds` -> on a `productListAdds`

File: what-to-implement/commerce.md [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+* [product](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/content/productlistitem.schema.md#xdmproduct) (_Highly Recommended_) -- The XDM id for the product.
+* [productAddMethod](https://github.com/adobe/xdm/blob/1c22180490558e3c13352fe3e0540cb7e93c69ca/docs/reference/content/productlistitem.schema.md#xdmproductaddmethod) (_Highly Recommended_) -- The method that was used to add a product item to the list by the visitor. Set with product list add metrics. Should only be used when a product is added to the list. Examples include `add to cart button`, `quick add` and `upsell`.
```
Comment (Aaronius) [inline what-to-implement/commerce.md]: In some of these cases, like "Set with product list add metrics.", it's hard to parse which words are names of things vs just sentence structure. I'm not sure what the appropriate solution would be. Maybe:

Set with product-list-add metrics.

or

Set with "product list add" metrics.

or

Set with `productListAdds` metrics.

or something like that?

File: what-to-implement/commerce.md (hunk 5 above)
Comment (dancingcactus) [reply what-to-implement/commerce.md]: Sorry. Are you saying it should have a space after the Colon or no space. I agree I need to be consistent just not sure what convention is.

Comment (Aaronius) [reply what-to-implement/commerce.md]: Sorry, yes, a space should be after the colon.
